engine/src/agents/mctsagenttruesight.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+  CrazyAra, a deep learning chess variant engine
+  Copyright (C) 2018       Johannes Czech, Moritz Willig, Alena Beyer
+  Copyright (C) 2019-2020  Johannes Czech
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+/*
+ * @file: mctsagenttruesight.cpp
+ * Created on 05.2021
+ * @author: BluemlJ
+ */
+
+#include <string>
+#include <thread>
+#include <fstream>
+#include "mctsagenttruesight.h"
+#include "../evalinfo.h"
+#include "../constants.h"
+#include "../util/blazeutil.h"
+#include "../manager/treemanager.h"
+#include "../manager/threadmanager.h"
+#include "../node.h"
+#include "../util/communication.h"
+#include "util/gcthread.h"
+
+
+MCTSAgentTrueSight::MCTSAgentTrueSight(NeuralNetAPI *netSingle, vector<unique_ptr<NeuralNetAPI>>& netBatches,
+                     SearchSettings* searchSettings, PlaySettings* playSettings):
+    MCTSAgent(netSingle, netBatches, searchSettings, playSettings)
+    {
+        
+    }
+
+MCTSAgentTrueSight::~MCTSAgentTrueSight()
+{
+    for (auto searchThread : searchThreads) {
+        delete searchThread;
+    }
+}
+
+string MCTSAgentTrueSight::get_name() const
+{   
+    string ret = "MCTSTrueSight-" + engineVersion + "-" + net->get_model_name();
+    
+    
+    return ret;
+}
+
+void MCTSAgentTrueSight::evaluate_board_state()
```

Comment:
I mean I only need to change one line of the method if I am correct. If we design a new small method called ``` clone_rootstate() ``` (or similar) in mctsagent and use this method instead of ``` rootState = unique_ptr<StateObj>(state->clone()); ``` in ```evaluate_board_state()```, it would be possible to only override the new ``` clone_rootState() ``` method instead of the complete evaluation method. 

What is your opinion on this?